### PR TITLE
Homework6: Vladimir Natsevsky

### PR DESCRIPTION
**Pull request by JunglistMNSQ**

### CODE REVIEW

File: src/homework6/decorator.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+исключение типа TooManyErrors
+
+"""
+
+
+def decorator(errors):
+
+    def dec(func):
+        def wrapper(*args, **kwargs):
+            nonlocal errors
```
Comment (belovmd) [inline src/homework6/decorator.py]: why it's nonlocal? we don't have it in enclosure scope

File: src/homework6/decorator.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            while errors:
+                try:
+                    result = func(*args, **kwargs)
+                except Exception:
+                    errors -= 1
+                    return 'Something wrong'
+                else:
+                    return result
+            else:
+                return 'TooManyErrors'
```
Comment (belovmd) [inline src/homework6/decorator.py]: it should be exception class

File: src/homework6/modelfromlife.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+                               8, 9, 10, 10, 10, 10
+                               ]
+        self.one_hand_bandit_ = [1, 2, 3, 4, 5, 6, 7]
+        self.start_capital = 10000
+        self.capital = 10000
+        self.roulette_players = []
+        self.blackjack_players = []
+        self.bar_visitors = []
+
+    def cycle(self, times):
```
Comment (belovmd) [inline src/homework6/modelfromlife.py]: it should be moved out of casino class at all

File: src/homework6/modelfromlife.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+                self.roulette()
+                self.blackjack(self.blackjack_players)
+                self.blackjack_players.clear()
+                self.bar()
+                self.bar_visitors.clear()
+                print(self)
+            else:
+                break
+
+    def roulette(self):
```
Comment (belovmd) [inline src/homework6/modelfromlife.py]: it should take players as parameter

File: src/homework6/modelfromlife.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+        while sum(croupier_hand) < 17:
+            croupier_hand.append(cards.pop(0))
+        for person in players:
+            if 22 > sum(person[2]) > sum(croupier_hand):
+                person[0].won(person[1] * 2)
+                self.capital -= person[1]
+            else:
+                self.capital += person[1]
+
+    def bar(self):
```
Comment (belovmd) [inline src/homework6/modelfromlife.py]: bar can be put to separate class and here it would be an attribute

File: src/homework6/modelfromlife.py [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+            self.current_place = random.choice(self.prefer_place)
+
+    def do_i_have_cash(self):
+        if self.money - (self.start_bet * self.drunk) <= 0:
+            self.gameover()
+            return False
+        else:
+            return True
+
+    def won(self, pot):
```
Comment (belovmd) [inline src/homework6/modelfromlife.py]: it should be moved to games (or to casino class)

**Review by belovmd — COMMENTED** [review]
please take a look at these issues

**Review by belovmd — COMMENTED** [review]
please address previous review issues

File: src/homework6/decorator.py (hunk 1 above)
Comment (JunglistMNSQ) [reply src/homework6/decorator.py]: cleared

File: src/homework6/decorator.py (hunk 2 above)
Comment (JunglistMNSQ) [reply src/homework6/decorator.py]: made class TooManyErrors

File: src/homework6/modelfromlife.py (hunk 3 above)
Comment (JunglistMNSQ) [reply src/homework6/modelfromlife.py]: moved out

File: src/homework6/modelfromlife.py (hunk 4 above)
Comment (JunglistMNSQ) [reply src/homework6/modelfromlife.py]: added parameter players

File: src/homework6/modelfromlife.py (hunk 5 above)
Comment (JunglistMNSQ) [reply src/homework6/modelfromlife.py]: separated

File: src/homework6/modelfromlife.py (hunk 6 above)
Comment (JunglistMNSQ) [reply src/homework6/modelfromlife.py]: moved to casino class

**Comment by JunglistMNSQ** [comment]
Fixed all issues

File: src/homework6/decorator.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                while errors:
+                    try:
+                        result = func(*args, **kwargs)
+                        return result
+                    except Exception:
+                        errors -= 1
+                        print('Something wrong')
+                else:
+                    raise TooManyErrors
+            except TooManyErrors:
```
Comment (belovmd) [inline src/homework6/decorator.py:28]: I don't understand why should we raise it, than at the same time catch it and then return exception class.

**Review by belovmd — COMMENTED** [review]
7
decorator isn't solved

**Pull request closed without merging**